CMakeLists.txt

```diff
@@ -34,6 +34,16 @@ project(
 #set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO ON CACHE STRING "")
 
 option(MRDOX_BUILD_TESTS "Build tests" ON)
+option(MRDOX_BUILD_SHARED "Link shared" OFF)
+
+if (MRDOX_BUILD_SHARED)
```

Comment:
is this how its usually done with cmake? I mean, is there not some kind of standard switch or something?